src/polyfill.rs

```diff
@@ -146,3 +146,18 @@ macro_rules! slice_as_array_ref_mut {
         }
     }
 }
+
+/// Returns a slice of arrays starting at the given pointer.
+macro_rules! ptr_as_array_slice {
+    ($ptr:expr, $block_len:expr, $block_num:expr) => {
```

Comment:
As long as `sha1::block_data_order` takes `data: *const u8, num: c::size_t`, calling this new API would require an unsafe `slice::from_raw_parts` call:

```
let data = core::slice::from_raw_parts(data, num * BLOCK_LEN);
let (blocks, rest) = slice_as_array_ref_chunks(data, BLOCK_LEN);
debug_assert!(rest.is_empty());
```

> do so in a way where the feature could reasonably be expected to make it into the Rust language or stdlib.

In my opinion this would be very tough sell for `slice_as_array_ref_chunks`. It’s a very narrow use case. Who knows how many slightly different functions like this would be needed to cover every use of `as` ever?